src/java.base/share/classes/java/util/FormatProcessor.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.util;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.invoke.MethodHandles.Lookup;
+import java.lang.template.ProcessorLinkage;
+import java.lang.template.StringProcessor;
+import java.lang.template.StringTemplate;
+import java.lang.template.ValidatingProcessor;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import jdk.internal.javac.PreviewFeature;
+
+/**
+ * This {@linkplain ValidatingProcessor template processor} constructs a {@link String}
+ * result using {@link Formatter}. Unlike {@link Formatter}, {@link FormatProcessor} uses
+ * the value from the embedded expression that follows immediately after the
+ * <a href="../util/Formatter.html#syntax">format specifier</a>.
+ * StringTemplate expressions without a preceding specifier, use "%s" by
+ * default. Example:
+ * {@snippet :
+ * int x = 10;
+ * int y = 20;
+ * String result = FMT."%05d\{x} + %05d\{y} = %05d\{x + y}";
+ * }
+ * result is: <code>00010 + 00020 = 00030</code>
+ *
+ * @implNote When used in conjunction with a runtime instances of {@link
+ * StringTemplate} representing string templates this {@link StringProcessor}
+ * will use the format specifiers in the fragments and types of the values in
+ * the value list to produce a more performant formatter.
+ *
+ * @implSpec Since, values are found within the string template, argument indexing
+ * specifiers are unsupported.
```

Comment:
What is the behavior of a format that references an index?  Ignored, throw?
I can see an argument for simplicity, but the values are a sequential list similar to that of a varargs call to format("xx", x, y, z, ...) so it can be well defined to what value an index in the format refers.

Reply:
Kind of blows the whole concept of embedded expressions. This is one of those, just because you can, doesn't mean you should.

Reply:
ok, still what is the behavior if the format includes an index?
The pattern scanner in FormatProcessor permits an index and does not cause an error.

Also, there might be a small benefit to using the existing compiled pattern in Formatter.java.  (make it package private and rename to uppercase).

Reply:
Currently it issues an `MissingFormatArgumentException` which I guess could be confusing.  Changing to IllegalFormatFlagsException with a descriptive message.

Re: Pattern - I was originally trying to minimize changes to Formatter, but considering what I did change...